plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -43,11 +43,14 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
     private val classes: MutableMap<MangledName, ClassTypeEmbedding> = mutableMapOf()
     private val properties: MutableMap<MangledName, PropertyEmbedding> = mutableMapOf()
 
-    override val anonNameProducer = FreshEntityProducer { AnonymousName(it) }
-    override val whileIndexProducer = FreshEntityProducer { it }
-    override val returnLabelNameProducer = FreshEntityProducer { ReturnLabelName(it) }
-    override val catchLabelNameProducer = FreshEntityProducer { CatchLabelName(it) }
-    override val tryExitLabelNameProducer = FreshEntityProducer { TryExitLabelName(it) }
+    override val anonNameProducer = simpleFreshEntityProducer { AnonymousName(it) }
+    override val whileIndexProducer = indexProducer()
+    override val catchLabelNameProducer = simpleFreshEntityProducer { CatchLabelName(it) }
+    override val tryExitLabelNameProducer = simpleFreshEntityProducer { TryExitLabelName(it) }
+    override val scopeIndexProducer = indexProducer()
+
+    // For some reason, type deduction does not seem to work here.
+    override val returnTargetProducer = FreshEntityProducer<ReturnTarget, TypeEmbedding> { n, type -> ReturnTarget(n, type) }
```

Comment:
Sometimes the type inference engine is not able to solve these type equations. If you want, you can substitute the type with FreshEntityProducer<_, TypeEmbedding>.